### PR TITLE
ctb: Add events and custom errors to LivenessModule

### PR DESCRIPTION
**Pull request by maurelian**
Add events and custom errors to the LivenessModule. The `RemoveOwner` event matches the one emitted by the Safe contract.

New functionality is tested.

## Summary by CodeRabbit

- **New Features**
	- Introduced new error messages for insufficient owners in the Safe contract.
	- Added events for owner removal and fallback owner taking ownership for enhanced observability.

- **Documentation**
	- Updated `semver-lock.json` with new `initCodeHash` and `sourceCodeHash` values for the `LivenessModule.sol` file.

- **Refactor**
	- Modified logic to emit specific events during owner removal and fallback owner actions in the Safe contract tests.

### CODE REVIEW

**Review by smartcontracts — APPROVED** [review]
LGTM, note that the PR title is outdated

File: packages/contracts-bedrock/src/Safe/LivenessModule.sol [hunk 1]
```diff
@@ -14,6 +14,12 @@ import { ISemver } from "src/universal/ISemver.sol";
 ///         If the number of owners falls below the minimum number of owners, the ownership of the
 ///         safe will be transferred to the fallback owner.
 contract LivenessModule is ISemver {
+    /// @notice Emitted when an owner is removed due to insufficient liveness
+    event RemovedOwner(address indexed owner);
+
+    /// @notice Emitted when the fallback owner takes ownership
+    event FallbackOwnerTakesOwnership();
```
Comment (mds1) [inline packages/contracts-bedrock/src/Safe/LivenessModule.sol]: Nit on tense, `RemovedOwner` is past tense and `FallbackOwnerTakesOwnership` is present tense

Comment (maurelian) [reply packages/contracts-bedrock/src/Safe/LivenessModule.sol]: OK, both are past tense now.

**Comment by maurelian** [comment]
> LGTM, note that the PR title is outdated

Yes, will change errors in another PR.

**Pull request merged**